### PR TITLE
Add sole proprietor info to get biz endpoint response

### PR DESCRIPTION
**Pull request by ruaraya**
Add sole proprietor info to get biz endpoint response. In addition, integrated tests were added to the related get service.

### CODE REVIEW

File: src/Spd.Resource.Repository/Biz/Mappings.cs [hunk 1]
```diff
@@ -93,5 +96,29 @@ private static string SetAddressField(string src, string dest)
...
             return src;
         }
+
+        private static SwlContactInfo GetSoleProprietorContactInfo(IEnumerable<spd_licence> licences)
+        {
+            spd_licence? solePropietorInfo = licences.FirstOrDefault(l => l.statuscode == DynamicsConstants.StatusCode_Active);
+            
+            return new SwlContactInfo()
+            {
+                BizContactId = solePropietorInfo?.spd_LicenceHolder_contact?.spd_contact_spd_businesscontact.FirstOrDefault()?.spd_businesscontactid,
```
Comment (peggy-quartech) [inline src/Spd.Resource.Repository/Biz/Mappings.cs:106]: BizContactId should be null, sole proprientor licence does not have bizContact

Comment (ruaraya) [reply src/Spd.Resource.Repository/Biz/Mappings.cs:106]: If the field is always null, why do we need to include it as part of the response?

Comment (peggy-quartech) [reply src/Spd.Resource.Repository/Biz/Mappings.cs:106]: no, in this situation, it is null.
The SwlContactInfo type will also be used in ControllingMember, Employee etc. At that scenario, it won't be null.

Comment (ruaraya) [reply src/Spd.Resource.Repository/Biz/Mappings.cs]: Got it, fixed.

**Comment by peggy-quartech** [comment]
I think you also need to add "sole proprietor" code for PUT endpoint (update profile)
and also validation for update.

File: src/Spd.Manager.Licence/BizProfileContract.cs [hunk 2]
```diff
@@ -25,14 +25,12 @@ public record BizProfileUpdateCommand(
 
 public record BizInfo
 {
-    public string? BizLegalName { get; set; }
```
Comment (peggy-quartech) [inline src/Spd.Manager.Licence/BizProfileContract.cs:28]: if you remove this, when we do read, where does user get BizLegalName?

File: src/Spd.Resource.Repository/Biz/BizRepository.cs [hunk 3]
```diff
@@ -46,23 +46,40 @@ public async Task<IEnumerable<BizResult>> QueryBizAsync(BizsQry qry, Cancellatio
...
             
-            if (Biz == null) throw new ApiException(HttpStatusCode.NotFound);
+            if (biz == null) throw new ApiException(HttpStatusCode.NotFound);
 
             List<spd_account_spd_servicetype> serviceTypes = _context.spd_account_spd_servicetypeset
-                .Where(so => so.accountid == Biz.accountid)
+                .Where(so => so.accountid == biz.accountid)
                 .ToList();
 
+            Guid? licHolderId = biz.spd_organization_spd_licence_soleproprietor
```
Comment (peggy-quartech) [inline src/Spd.Resource.Repository/Biz/BizRepository.cs:63]: this is actually licenceId

File: src/Spd.Resource.Repository/Biz/BizRepository.cs [hunk 4]
```diff
@@ -46,23 +46,40 @@ public async Task<IEnumerable<BizResult>> QueryBizAsync(BizsQry qry, Cancellatio
...
+                ._spd_licenceholder_value;
+            spd_licence? licHolderContact = new();
+
+            if (licHolderId != null && licHolderId != Guid.Empty)
+                licHolderContact = _context.spd_licences
+                    .Expand(l => l.spd_LicenceHolder_contact)
+                    .Where(l => l.statecode != DynamicsConstants.StateCode_Inactive)
+                    .Where(l => l._spd_licenceholder_value == licHolderId)
+                    .FirstOrDefault();
+
```
Comment (peggy-quartech) [inline src/Spd.Resource.Repository/Biz/BizRepository.cs:74]: We do not need this. 
 biz.spd_organization_spd_licence_soleproprietor
                .FirstOrDefault(l => l.statecode == DynamicsConstants.StateCode_Active)?
                ._spd_licenceholder_value
is the response.SoleProprietorSwlContactInfo.ContactId

 biz.spd_organization_spd_licence_soleproprietor
                .FirstOrDefault(l => l.statecode == DynamicsConstants.StateCode_Active)?
                .spd_licenceId
is response.SoleProprietorSwlContactInfo.LicenceId

Comment (ruaraya) [reply src/Spd.Resource.Repository/Biz/BizRepository.cs]: Fixed.

**Review by peggy-quartech — APPROVED** [review]

**Pull request merged**